app/models/solidus/gateway/braintree_gateway.rb

```diff
@@ -254,7 +255,9 @@ def transaction_authorize_or_purchase_params(cents, creditcard, options = {})
       params[:options] ||= {}
       params[:amount] = amount(cents)
       params[:channel] ||= "Solidus"
-      params[:shipping] = map_address(options[:shipping_address]) if options[:shipping_address]
+      params[:shipping] = map_address(options[:shipping_address]) if (
+        options[:shipping_address] && preferred_transmit_shipping_address
+      )
```

Comment:
I would prefer to see this as a standard `if` statement, rather than this multi-line version of the single-line conditional pattern.